### ISSUE TITLE
History matching and control mode data represented in same objects

### ISSUE THREAD

**Issue by sveinung-r** [OPM/opm-parser#1163]
WellProductionProperties and WellInjectionProperties are currently representing both history matching and control mode data. This may create some confusion as to the meaning of terms such as BHP, OilRate etc. as these may be limits set for simulation or historic data. Having separate classes would enable inferring the meaning of these terms from context.

Perhaps these should be split in different types representing historic and control mode data separately?

**Comment by GitPaean** [comment]
Yes. We have discussed in https://github.com/OPM/opm-parser/pull/1032 .  Thanks for pointing it out though. We need to address it.

**Comment by GitPaean** [comment]
We will try to fix it, it also it needs to be fixed from the simulator side and some work in opm-core.
